Dockerfile

```diff
@@ -0,0 +1,89 @@
+FROM ruby:2.5-slim
+LABEL authors="Rob Wilkinson <rob@oddball.io>, Keifer Furzland <keifer@oddball.io>"
+
+# Set some variables
+ENV REPO_DIR /home/magnifier/src
+ENV GEM_DIR /home/magnifier/gems
+
+# Set application specific environment variables
+ENV PORT=3000
+ENV RAILS_ENV='production'
+ENV RACK_ENV='production'
+ENV NODE_ENV='production'
+ENV RAILS_SERVE_STATIC_FILES='true'
+ENV PATH="`ruby -e 'puts Gem.user_dir'`/bin:$PATH"
+
+# Inject RAILS_MASTER_KEY
+ARG RAILS_MASTER_KEY
+ENV RAILS_MASTER_KEY=$RAILS_MASTER_KEY
+
+# We don't need documentashun
+RUN echo "gem: --no-ri --no-rdoc" | tee /etc/gemrc
+
+# Install curl so we can install yarn (ruby-slim doesn't have curl)
+RUN apt-get update -yqq && \
+    apt-get install -yqq --no-install-recommends curl gnupg2 && \
+    curl -sL https://deb.nodesource.com/setup_10.x | bash - && \
+    curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - && \
+    echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list
```

Comment:
One capability that works really well for situations where you need to download/install stuff that you don't want to be in the final image is [multistage builds](https://docs.docker.com/develop/develop-images/multistage-build/). Although I've only used them when building go applications and I only had to copy over a single binary to the last stage. I'm not sure how great it would be with rails.

Anyways a multistage build is basically multiple dockerfiles in one and you can copy things from one to another. So, basically, in go you would install a ton of dependencies and build the binary in the first image and in the second image copy the binary over and you are left with that last image that contains just the binary.

Again, I don't know rails very well so it may not be a good fit but I thought I'd throw it out there.

Reply:
I think this is actually a perfect fit - I've read about multistage builds but hadn't gotten around to it. I think for this particular PR I'll probably forgo that work effort, but it's a good opportunity for a chunk of refactoring in the future